crates/yakui-core/src/paint/paint_dom.rs

```diff
@@ -203,6 +204,10 @@ impl PaintDom {
             let mut pos = vertex.position * self.scale_factor;
             pos += self.unscaled_viewport.pos();
             pos /= self.surface_size;
+            if mesh.pipeline == Pipeline::Text {
```

Comment:
Why does this only apply to the text pipeline? Wouldn't other textures also be affected here?

If only text is affected, can we round the vertex positions in `text_renderer` instead of special casing the pipeline here? I think this is too late in the rendering pipeline to have code that specializes based on the pipeline.

Reply:
The reason it's in this stage of the pipeline is that during it we also multiply it by the scale factor, and if we remember correctly, RoundRect depends on this subpixel position to behave correctly

Reply:
We might need to figure out how to move the scale factor multiplication out of the pipeline if we wanted to resolve this

Reply:
Ah, gotcha. We need to round to the nearest physical pixel, but components output vertices in logical pixels, where it's too early in the pipeline to round.

I'd be surprised if any components relied on sub-physical-pixel vertex positions! If we can find a widget in the tree that gets worse without subpixel positions, I'm sold.

Reply:
For rounded rect:
first: rounded
second: unrounded

we think this issue is even more pronounced with a non-integer scale factor

![image](https://github.com/SecondHalfGames/yakui/assets/33348966/e1bc9c09-6863-4efc-b537-f9c6023a7d75)
![image](https://github.com/SecondHalfGames/yakui/assets/33348966/d3b538e9-899d-4ceb-8ba7-0aa2d38bf675)

Reply:
![image](https://github.com/SecondHalfGames/yakui/assets/33348966/188476cd-06bf-44b5-8d29-2a512e1fb6ee)

rounded, with a scale factor of 1.25